### PR TITLE
Introducing dialected-filters to the contract for supporting new-trigger-filters experimental feature

### PR DESCRIPTION
**Pull request by devguyio**
Part of #1763 

This PR introduces dialected-filters to the egress contract as a first step of implementing the new-trigger-filers experimental feature.

## Proposed Changes

- Add dialected filters to egress contract

### CODE REVIEW

**Review by matzew — APPROVED** [review]
/lgtm
/approve

I like the commit message details

File: proto/contract.proto [hunk 1]
```diff
@@ -11,6 +11,48 @@ option java_outer_classname = "DataPlaneContract";
...
+
+message Prefix {
+  string attribute = 1;
+  string prefix = 2;
+}
+
+message Suffix {
+  string attribute = 1;
+  string suffix = 2;
+}
```
Comment (pierDipi) [inline proto/contract.proto:26]: I'm reading [this example in spec](https://github.com/cloudevents/spec/blob/main/subscriptions/spec.md#filters):
```
[ {"prefix": { "type": "com.github.issue" } } ]
```

why doesn't this need to be a `map<string, string>`?

In addition, at this layer, I would design for extensibility here.

Comment (pierDipi) [reply proto/contract.proto:26]: I see why:
> Use of this MUST include exactly one nested property

It seems weird that exact supports multiple attributes while prefix and suffix don't

**Pull request merged**